src/ServiceControl.Transports.SqlServer/QueueLengthProvider.cs

```diff
@@ -38,36 +35,27 @@ public void TrackEndpointInputQueue(EndpointToQueueMapping queueToTrack)
             tableSizes.TryAdd(sqlTable, 0);
         }
 
-        public Task Start()
+        protected override async Task ExecuteAsync(CancellationToken stoppingToken)
         {
-            stop = new CancellationTokenSource();
-
-            poller = Task.Run(async () =>
+            while (!stoppingToken.IsCancellationRequested)
```

Comment:
Would the `PeriodicTimer` be a better fit for this background task?
I used it in https://github.com/Particular/ServiceControl/blob/1b522c5bd43c7ea55a0025720ccdad778affcc7e/src/Particular.LicensingComponent/BrokerThroughput/BrokerThroughputCollectorHostedService.cs#L34 and feels nicer.
I am just wondering.

Reply:
And yes, I noticed this is existing code and they are all the same pattern!

Reply:
Yeah the reason I use the while with the delay pattern is because the existing code was already doing that. At least from that perspective I wanted to stay compatible. I was aware the period timer pattern exists. It is also documented here https://learn.microsoft.com/en-us/aspnet/core/fundamentals/host/hosted-services?view=aspnetcore-8.0&tabs=visual-studio#asynchronous-timed-background-task

Can you elaborate on the "feels nicer" part?